Add challenge36: Advanced reverse engineering game (#947)
OWASP/wrongsecrets: src/main/resources/explanations/challenge36_reason.adoc

*Why should we be careful with the constants declared in the program?*

> Constants are fixed values that are useful for storing values that don't need to change during program execution. During the reverse engineering process, binary exploration is a technique that consists of extracting the maximum amount of information from the binary, in order to understand the logic of the program. However, some crucial information must be encrypted in such a way as to make the crucial information difficult to read.

Reviewer: ```suggestion
Constants are fixed values that are useful for storing values that don't need to change during program execution. During the reverse engineering process, binary exploration is a technique that consists of extracting the maximum amount of information from the binary, in order to understand the logic of the program. However, some crucial information must be encrypted in such a way as to make the crucial information difficult to read.

As you can tell by now: even encryption will just slow an attacker down, but not block him from finding the crucial information. Therefore: consider putting the actual crucial information online instead of only secured by offline encryption.
```

> *Why should we be careful with the constants declared in the program?*
> 
> Constants are fixed values that are useful for storing values that don't need to change during program execution. During the reverse engineering process, binary exploration is a technique that consists of extracting the maximum amount of information from the binary, in order to understand the logic of the program. However, some crucial information must be encrypted in such a way as to make the crucial information difficult to read.
> 
> As you can tell by now: even encryption will just slow an attacker down, but not block him from finding the crucial information. Therefore: consider putting the actual crucial information online instead of only secured by offline encryption.

Reviewer 2: ```suggestion
*Why should we be careful with the constants declared in the program?*

Constants are fixed values useful for storing values that don't need to change during program execution. During the reverse engineering process, binary exploration is a technique that consists of extracting the maximum amount of information from the binary to understand the logic of the program. However, some crucial information must be encrypted in such a way as to make said information challenging to read.

This shows that even encryption will slow an attacker down but not block him from finding the crucial information. Therefore, consider putting the crucial information online instead of only secured by offline encryption.
```

> As you can tell by now: even encryption will just slow an attacker down, but not block him from finding the crucial information. Therefore: consider putting the actual crucial information online instead of only secured by offline encryption.

Reviewer 2: > Therefore: consider putting the actual crucial information online instead of only secured by offline encryption.

I don't really get this sentence. Could you explain?

Reviewer: The idea by the challenge is that it is better to authenticate towards a server in order to get a secret, than to have the secret offline encrypted: as you can eventually still get it.
